internal/icon/xpm.go

```diff
@@ -0,0 +1,125 @@
+package icon
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"strconv"
+	"strings"
+)
+
+func parseXPM(data []byte) image.Image {
+	var colCount, charSize int
+	colors := make(map[string]color.Color)
+	var img *image.NRGBA
+
+	rowStart := 0
+	rowNum := 0
+	for i, b := range data {
+		if b == '\n' {
+			row := string(data[rowStart:i])
+			rowStart = i + 1
+			if row == "" || row[0] != '"' {
+				continue
+			}
+			row = stripQuotes(row)
+
+			if rowNum == 0 {
+				w, h, cols, size := parseDimensions(row)
+				img = image.NewNRGBA(image.Rectangle{image.Point{}, image.Point{w, h}})
+				colCount = cols
+				charSize = size
+			} else if rowNum <= colCount {
+				id, c := parseColor(row)
+				if id != "" {
+					colors[id] = c
+				}
+			} else {
+				parsePixels(row, charSize, rowNum-colCount-1, colors, img)
+			}
+			rowNum++
+		}
+	}
+
+	return img
+}
+
+func parseColor(data string) (id string, c color.Color) {
+	if len(data) == 0 {
+		return
+	}
+	parts := strings.Fields(data)
+	if len(parts) != 3 {
```

Comment:
This seems like an unnecessary if statement since the next line check if len(parts) == 2

Reply:
I don't follow. There are 3 cases handled.
* 3 parts (skips)
* 2 parts (adds the "c")
* other (probably error, just returns)

Reply:
so instead of nesting the whole if/else and the important code, just nest the return in the else
if len(parts) == 2 && parts[0] == "c" {
        parts = []string{" ", "c", parts[1]}
} else {
        if len(parts) != 3 {
	        return
        }
}

Reply:
As indicated there it is moved rather than simplified as there is still the double nesting.
However switch that to an "else if" and you have a point. Updating.